src/EditorFeatures/Core/Implementation/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.TagProducer.cs

```diff
@@ -56,6 +56,10 @@ public void Dispose()
                 }
 
                 var document = snapshotSpans.First(vt => vt.SnapshotSpan.Snapshot == position.Snapshot).Document;
+                if (document == null)
+                {
+                    return SpecializedTasks.EmptyEnumerable<ITagSpan<AbstractNavigatableReferenceHighlightingTag>>();
+                }
```

Comment:
I'm not clear on why this `NullReferenceException` happens in the first place. Can you provide some context?